File: apps/mobile/ios/kini/KiniNativeSwitchManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(KiniNativeSwitchManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(checked, BOOL)
RCT_EXPORT_VIEW_PROPERTY(disabled, BOOL)
RCT_EXPORT_VIEW_PROPERTY(onChange, RCTBubblingEventBlock)

@end

File: apps/mobile/ios/kini/KiniNativeSwitchManager.swift
import React
import UIKit

@objc(KiniNativeSwitchManager)
final class KiniNativeSwitchManager: RCTViewManager {
  override func view() -> UIView! {
    KiniNativeSwitchView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    true
  }
}
